runner/serializers.py

```diff
@@ -149,19 +149,16 @@ def create(self, validated_data):
         tags = validated_data.get('tags')
         create_date = datetime.datetime.now().strftime("%m/%d/%Y, %H:%M:%S")
         name = "Run %s: %s" % (pipeline.name, create_date)
-        output_directory = validated_data.get('output_directory')
         if validated_data.get('name'):
             name = validated_data.get('name') + ' (' + create_date + ')'
-        if validated_data.get('output_directory'):
-            output_directory = validated_data.get('output_directory',
-                                                  os.path.join(pipeline.output_directory, str(self.id)))
         run = Run(name=name,
                   app=pipeline,
                   status=RunStatus.CREATING,
                   job_statuses=dict(),
                   output_metadata=validated_data.get('output_metadata', {}),
-                  tags=tags,
-                  output_directory=output_directory)
+                  tags=tags)
+        if validated_data.get('output_directory'):
+            run.output_directory=validated_data.get('output_directory')
```

Comment:
You can avoid if statement by doing `run.output_directory=validated_data.get('output_directory', None)`.